docs/guides/aerogear-android/push.asciidoc

```diff
@@ -89,100 +116,46 @@ You can also add a default message handler to the receiver in AndroidManifest.xm
-
-[source,java]
-----
-registrar.unregister(getApplicationContext(), new Callback<Void>() {
     @Override
-    public void onSuccess(Void ignore) {
-        Toast.makeText(getApplicationContext(), "Unregistration succeeded", Toast.LENGTH_LONG).show();
-        Log.i(TAG, "SUCCESS");
+    public void onMessage(Context context, Bundle bundle) {
+        TextView text = (TextView) findViewById(R.id.text_view1);
+        text.setText(bundle.getString("alert"));
+        text.invalidate();
     }
 
     @Override
-    public void onFailure(Exception exception) {
-        Toast.makeText(getApplicationContext(), "Unregistration failed", Toast.LENGTH_LONG).show();
-        Log.e(TAG, exception.getMessage(), exception);
+    public void onDeleteMessage(Context context, Bundle bundle) {
+        //ignore
     }
-});
-
-----
-
-==== Handling messages
-
-Currently, push messages are received by an instance of link:/docs/specs/aerogear-android/org/jboss/aerogear/android/unifiedpush/AeroGearGCMMessageReceiver.html[AeroGearGCMMessageReceiver]. They are processed and passed to link:/docs/specs/aerogear-android/org/jboss/aerogear/android/unifiedpush/Registrations.html[Registrations] via notifyHandlers method.
-
-Classes which implement MessageHandler can ask that their methods be called on either the UI thread using link:/docs/specs/aerogear-android/org/jboss/aerogear/android/unifiedpush/Registrations.html#registerMainThreadHandler(org.jboss.aerogear.android.unifiedpush.MessageHandler)[Registrations.registerMainThreadHandler] or on a background thread using link:/docs/specs/aerogear-android/org/jboss/aerogear/android/unifiedpush/Registrations.html#registerBackgroundThreadHandler(org.jboss.aerogear.android.unifiedpush.MessageHandler)[Registrations:registerBackgroundThreadHandler]. UI threads are most useful for Activities and Fragments, background threads are most useful for everything else.
 
-In an Activity you MUST remove the handler when the Activity goes into the background and MUST reenable it when it comes into the foreground.
-
-[source,java]
-----
-public class MainActivity extends Activity implements MessageHandler {
-
-	@Override
-	protected void onCreate(Bundle savedInstanceState) {
-		setContentView(R.layout.activity_main);
-	}
-
-	@Override
-	protected void onResume() {
-		super.onResume();
-                Registrations.registerMainThreadHandler(this);
-	}
-
-	@Override
-	protected void onPause() {
-		super.onPause();
-		Registrations.unregisterMainThreadHandler(this);
-	}
-
-	@Override
-	public void onMessage(Context context, Bundle bundle) {
-		TextView text = (TextView) findViewById(R.id.text_view1);
-		text.setText(bundle.getString("alert"));
-		text.invalidate();
-	}
-
-	@Override
-	public void onDeleteMessage(Context context, Bundle bundle) {
-		//ignore
-	}
-
-	@Override
-	public void onError() {
-		//ignore
-	}
+    @Override
+    public void onError() {
+        //ignore
+    }
 
 }
 ----
```

Comment:
In the push section I would rather link to the push examples (which aren't part of the cook book) instead of the cookbook.

The cookbook should have a link to both, I'll check and file a JIRA.